src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { ReactComponent as CheckmarkNormal } from './CheckmarkNormal.svg';
+import { ReactComponent as CheckmarkCompact } from './CheckmarkCompact.svg';
+import classes from './Checkbox.module.css';
+
+export interface CheckboxProps {
+  checkboxId?: string;
+  checked?: boolean;
+  className?: string;
+  compact?: boolean;
+  disabled?: boolean;
+  error?: boolean;
+  label?: string;
+  name?: string;
+  readOnly?: boolean;
+}
+
+export const Checkbox = ({
+  checkboxId,
+  checked,
+  className,
+  compact,
+  disabled,
+  error,
+  label,
+  name,
+  readOnly,
+}: CheckboxProps) => {
+  const [isChecked, setIsChecked] = useState<boolean>(!!checked);
+
+  useEffect(() => setIsChecked(!!checked), [checked]);
```

Comment:
I think it would be better to expose a `onChange` callback to the consumer, and leave the component controlled instead. By doing that, we dont need to react to the `checked` property, and update some internal state via a `useEffect`. This quickly gets a bit confusing.

Reply:
The problem is that the wrapper needs to know the state of the checkbox in order to apply correct styling. Is there a better way of doing that? I considered using the `:has()` pseudo-class in CSS, but it is not widely supported yet.

Reply:
We can still style based on the `checked` property, as it will still be a prop. The difference is that the consumer is responsible for controlling and updating the state, based on some callback to trigger the toggle.

Reply:
Okay, now I understand. It will require more from the consumers' part though, but i understand that this is mostly how they will use it anyway.